cmds/core/cat/cat_test.go

```diff
@@ -59,3 +68,53 @@ func TestCat(t *testing.T) {
 		t.Fatalf("Reading files failed: got %v, want %v", b.Bytes(), someData)
 	}
 }
+
+func TestCatPipe(t *testing.T) {
+	var inputbuf bytes.Buffer
+	teststring := "testdata"
+	fmt.Fprintf(&inputbuf, "%s", teststring)
+	osStdin = &inputbuf
+
+	var out bytes.Buffer
+	osStdout = &out
+
+	emptyargs := make([]string, 0)
+	if err := cat(&out, emptyargs); err != nil {
+		t.Error(err)
+	}
+	if out.String() != teststring {
+		t.Errorf("Input string: %s and output: '%s' string are not equal", teststring, out.String())
```

Comment:
If you print them with %q it will do the '' for you
If you can follow the standard pattern
t.Errorf("CatPipe: got %q, want %q", teststring, out.String()), would be nice
overall I'm a bit uncomfortable with the globals, it requires the resetGlobal function.

The u-root test framework supports spawning processes and running them. I'd like to hear from Ryan and get his thoughts, but think I'd prefer two things: 
func cat should take an io.Writer and io.Reader... and foreach io.Reader copy it to the io.Writer

For simple tests, the test can call cat function; for full tests, the tests should spawn cat processes and test that way, rather than calling the cat function